tests/dualstack_test.py

```diff
@@ -0,0 +1,189 @@
+    v4_server_runner: _KubernetesServerRunner = None
+    v6_server_runner: _KubernetesServerRunner = None
+
+    @staticmethod
+    @override
+    def is_supported(config: skips.TestConfig) -> bool:
+        if config.client_lang in _Lang.CPP | _Lang.PYTHON | _Lang.JAVA:
+            return config.version_gte("v1.66.x")
+        return False
+
+    @classmethod
+    def setUpClass(cls):
+        """Force the canonical test server for all languages."""
+        ## TODO: when adding other languages: make sure the java image version is switched to the one with all features we need for the dualstack test
+        super().setUpClass()
+        if cls.lang_spec.client_lang is not _Lang.JAVA:
+            cls.server_image = xds_k8s_flags.SERVER_IMAGE_CANONICAL.value
+
+    def setUp(self):
+        super().setUp()
+        runner_args = dict(
+            image_name=self.server_image,
+            gcp_service_account=self.gcp_service_account,
+            td_bootstrap_image=self.td_bootstrap_image,
+            gcp_project=self.project,
+            app_label=_SERVERS_APP_LABEL,
+            gcp_api_manager=self.gcp_api_manager,
+            xds_server_uri=self.xds_server_uri,
+            network=self.network,
+            debug_use_port_forwarding=self.debug_use_port_forwarding,
+            reuse_namespace=True,
+            reuse_service=True,
+            enable_dualstack=True,
+            enable_workload_identity=self.enable_workload_identity,
+        )
+
+        k8s_namespace = self.server_runner.k8s_namespace
+        # Replaces self.server_runner initiated in self.initKubernetesServerRunner().
+        self.server_runner = _KubernetesServerRunner(
+            k8s_namespace,
+            deployment_name=self.server_name,
+            **runner_args,
+        )
+        self.v4_server_runner = _KubernetesServerRunner(
+            k8s_namespace,
+            deployment_name=self.server_name + "-v4",
+            service_name=self.server_runner.service_name,
+            **runner_args,
+        )
+        self.v6_server_runner = _KubernetesServerRunner(
+            k8s_namespace,
+            deployment_name=self.server_name + "-v6",
+            service_name=self.server_runner.service_name,
+            **runner_args,
+        )
+
+    def cleanup(self):
+        self.td.cleanup(force=self.force_cleanup)
+        self.client_runner.cleanup(
+            force=self.force_cleanup, force_namespace=self.force_cleanup
+        )
+        self.server_runner.cleanup(
+            force=self.force_cleanup, force_namespace=False
+        )
+        if self.v4_server_runner:
+            self.v4_server_runner.cleanup(
+                force=self.force_cleanup, force_namespace=False
+            )
+        if self.v6_server_runner:
+            self.v6_server_runner.cleanup(
+                force=self.force_cleanup, force_namespace=True
+            )
+
+    def test_dual_stack(self) -> None:
```

Comment:
noticed this too - let's make it `test_dual_stack` for consistency
```suggestion
    def test_dualstack(self) -> None:
```